frontend/src/metabase/dashboard/components/add-card-sidebar/QuestionPicker.jsx

```diff
@@ -49,7 +49,10 @@ function QuestionPicker({ onSelect, collectionsById, getCollectionIcon }) {
 
   const handleSearchTextChange = e => setSearchText(e.target.value);
 
-  const collections = (collection && collection.children) || [];
+  const allCollections = (collection && collection.children) || [];
+  const collections = isPublicCollection(dashboardCollection)
```

Comment:
From the spec, this only hides all personal collections when the dashboard is in a public collection.

1. Add a question to a dashboard in a public collection
    **expectation**:
    - hides all personal collections
    - show all questions
1. Add a question to a dashboard in a personal collection
    **expectation**:
    - shows all collections
    - show all questions
1. move a dashboard from a personal collection to a public collection and test 1)
    **expectation**: hides all personal collections
    - hides all personal collections
    - show all questions
1. move a dashboard from a public collection to a personal collection and test 2)
    **expectation**:
    - shows all collections
    - show all questions